### ISSUE TITLE
get bad_alloc in removeValueById

### ISSUE THREAD

**Issue by fari-99** [seqan/seqan#994]
hello, i want to remove Value in my StringSet. but i get bad_alloc.

here is my example code.

```(cpp)
#include <iostream>

#include <seqan/sequence.h>
#include <seqan/file.h>
#include <seqan/index.h>
#include <seqan/align.h>
#include <seqan/seq_io.h>

using namespace seqan;
using namespace std;

int main()
{
    StringSet<DnaString> sekuens;
    vector<int> positionWrong;
    for(int i = 0; i < 12; i++)
    {
        if(i%2 == 0)
        {
            appendValue(sekuens, "ACGT");
        }
        else
        {
            appendValue(sekuens, "TGCA");
        }
    }

    for(int j = 0; j < 12; j++)
    {
        if(j%2 == 0)
        {
            positionWrong.push_back(j);
        }
    }

    for(int k = 0; k < length(positionWrong); k++)
    {
        cout << positionWrong.at(k) << endl;
        cout << positionWrong.at(k) - k << endl;
        int pos = positionWrong.at(k) - k;
        removeValueById(sekuens,pos);
        cout << "remove success" << endl;
    }


    return 0;
}
```

in this code i try to remove value "ACGT" (in j%2 == 0) in my stringSet. i get the output like this.

```
0
0
remove success
2
1
/usr/include/seqan/basic/basic_exception.h:368 FAILED!  
   (Uncaught exception of type std::bad_alloc: std::bad_alloc)
```

thanks for helping,

best regards.

**Comment by h-2** [comment]
inserting and removing elements in stringsets has some issues, see also #973 .
If you are using a regular owner stringset like in the example, you could try working on the strings member of the stringset as a workaround, i.e. `removeValueById(sekuens.strings, pos);`.

**Comment by fari-99** [comment]
thanks its help alot.

2015-06-03 17:44 GMT+07:00 Hannes Hauswedell <notifications@github.com>:

**Comment by h-2** [comment]
I will leave this issue open until stringsets are fixed.

**Comment by fari-99** [comment]
thank you for your hard work :). i hope it done nicely. hehe

2015-06-04 16:59 GMT+07:00 Hannes Hauswedell <notifications@github.com>:

**Comment by h-2** [comment]
closed in favor of #973
